enhancements/dns/dns-operator-operand-logging-level.md

```diff
@@ -0,0 +1,230 @@
+---
+title: dns-operator-operand-logging-level
+authors:
+  - "@miheer"
+reviewers:
+  - "@alebedev87"
+  - "@candita"
+  - "@frobware"
+  - "@knobunc"
+  - "@Miciah"
+  - "@rfredette"
+approvers:
+  - "@frobware"
+  - "@knobunc"
+  - "@Miciah"
+  - "@alebedev87"
+  - "@rfredette"
+  - "@candita"
+creation-date: 2021-10-14
+last-updated: 2021-10-14
+status: implementable
+---
+
+# DNS Log Level API
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement describes the API and code changes necessary to expose
+a means to change the DNS Operator and CoreDNS Logging Levels to
+cluster administrators.
+
+## Motivation
+
+Supporting a trivial way to raise the verbosity of the DNS Operator and it's Operands (CoreDNS) would make debugging
+cluster dns operator and CoreDNS issues easier for cluster administrators and OpenShift developers.
+
+To have the CoreDNS enable the following classes such as error, denial and all.
+denial: either NXDOMAIN or nodata responses (Name exists, type does not). A nodata response sets the return code to NOERROR.
+error: SERVFAIL, NOTIMP, REFUSED, etc. Anything that indicates the remote server is not willing to resolve the request.
+all: the default - nothing is specified. Using of this class means that all messages will be logged whatever we mix together with "all".
+
+Also, a logging level API for the DNS Operator would assist OpenShift developers working on the DNS Operator who may
+desire more in-depth logging statements when working on the operator's controllers.
+
+Additionally, a logging level API for CoreDNS logs would assist cluster administrators who wish to have more control
+over the output of their DNSController's CoreDNS logs.
```

Comment:
This section would flow better if you swapped these two paragraphs (adjusting transitions as necessary).